00 Core/MWSE/mods/tew/AURA/Ambient/Interior/interiorToExterior.lua

```diff
@@ -0,0 +1,146 @@
+local data = require("tew.AURA.Ambient.Interior.interiorData")
+local config = require("tew.AURA.config")
+local sounds = require("tew.AURA.sounds")
+local common = require("tew.AURA.common")
+local cellData = require("tew.AURA.cellData")
+
+local moduleName = "interiorToExterior"
+local debugLog = common.debugLog
+
+local exteriorTimer, cellLast
+
+local function getEligibleCellType(cellType, actorCount)
+    if cellType then
+        if (data.names[cellType] or data.tavernNames[cellType])
+        and (actorCount) and (actorCount < 2) then
+            --debugLog(string.format("Too few people inside for interior type %s: %s", cellType, actorCount))
+            return nil
+        end
+        return cellType
+    end
+end
+
+local function cellCheck()
+    local mp = tes3.mobilePlayer
+    if (not mp) or (mp and (mp.waiting or mp.traveling)) then return end
+
+    exteriorTimer:pause()
+
+    local cell = tes3.getPlayerCell()
+    if not cell then return end
+
+    if (cell.isInterior)
+    or (cellLast and cellLast.isInterior and cell.isOrBehavesAsExterior) then
+        table.clear(cellData.exteriorDoors)
+    end
+
+    local modData = tes3.player.data.AURA
+    if not modData or not modData.visitedInteriorCells then return end
+
+    debugLog("Searching for eligible doors.")
+
+    for door in cell:iterateReferences(tes3.objectType.door) do
+        if not (door.destination and door.destination.cell.isInterior and door.tempData) then
+            goto nextDoor
+        end
+        local cellId = door.destination.cell.id:lower()
+        local visitedCellData = modData.visitedInteriorCells[cellId]
+        if (visitedCellData) and (visitedCellData.type) and (visitedCellData.lastVisited) then
+            local now = tes3.getSimulationTimestamp(true)
+            local last = visitedCellData.lastVisited
+
+            debugLog("Parsing door destination cell: " .. cellId)
+            debugLog(string.format("Last time visited: %.5f game hours ago.", (now - last)))
+
+            if not door.tempData.tew then door.tempData.tew = {} end
```

Comment:
Please please please if you could namespace it a bit deeper. It's unlikely I will reuse door ref's temp data, but who knows? Something like `tempData.tew.AURA.intToExt` or similar (maybe simpler).

Reply:
Right, makes sense.